battle.go

```diff
@@ -131,16 +133,29 @@ func (b *Battle) SimulateRound() []Transaction {
 				Move:   user.Moves[t.Move],
 				Damage: damage,
 			})
+		case ItemTurn:
+			receiver := b.getPokemon(t.Target.party, t.Target.partySlot)
+			move := receiver.Moves[t.Move]
+			transactions = append(transactions, ItemTransaction{
+				Target: receiver,
+				Item:   t.Item,
+				Move:   move,
+			})
 		default:
 			log.Panicf("Unknown turn of type %v", t)
 		}
 	}
-
 	// process transations
 	for _, transaction := range transactions {
 		switch t := transaction.(type) {
 		case DamageTransaction:
 			(*t.Target).CurrentHP -= t.Damage
+		case ItemTransaction:
+			if t.Item.Category == ItemPPRecovery {
+				t.Item.UseMoveItem(t.Target, t.Move)
+			} else {
+				t.Item.UseItem(t.Target)
+			}
```

Comment:
I'd prefer if the logic for processing the transaction was put here instead of using methods that are somewhere else. It would be better if public facing methods created a transaction that the battle processes. That way we have only one possible way to modify the state. And if we play back all the transactions for a battle in order, the resulting battle state is guaranteed to be the same every time, even if the user wants to cheat (outside the context of an `Agent` ofc).

Also, this feels backwards, like it should be `t.Target.UseItem(t.Item)` or `t.Move.UseItem(t.Item)`. This would also make more sense for when a pokemon auto-eats a berry

Reply:
There's a lot of items with a ton of different effects. If they don't get delegated to different methods, this file will become excessively long as its purpose is not to handle dealing with items.

I don't understand what you mean by "public facing methods created a transaction". Do you mean something like: `pokemon.UseItem(item *Item) Transaction`?

I do agree with the naming order, I will change that.